### PR TITLE
Add section about stack handling differences to known differences docs

### PR DESCRIPTION
**Pull request by FloThinksPi**
Add section about stack handling to known-differences-with-cf-for-vms.md

Feel free to correct me about that but to my understanding the current restage/staging behaviour in korifi comes with the difference to CF on VMs below, just to get that written down somewhere as @beyhan pointed me to this doc.

It might make sense to rephrase that by someone with deeper technical knowledge in the Korifi implementation and fact check it.

## Is there a related GitHub Issue?
No

## What is this change about?
Documentation

## Does this PR introduce a breaking change?
No

## Acceptance Steps
Crosscheck documentation

## Tag your pair, your PM, and/or team
None

### CODE REVIEW

**Comment by danail-branekov** [comment]
Without looking at the code and having experimented with that, here are my two cents.

As far as I know, whenever the stack on the kpack cluster builder changes, (**I think**) kpack schedules a new kpack build for each app. Once the app image is pushed, it would be set on the CFApp resource as a current droplet. Thus, whenever the app restarts, it would pick up the new version.
TL;DR - we need to experiment to see whether a restage is actually required, or a restart is enough.

The assumptions above is of course true when the default kpack bulder is used as a build runner. A custom build runner could behave differently.

**Comment by danail-branekov** [comment]
Hey @FloThinksPi 

I ran some experiments myself, and TL;DR what you suggest is completely correct.

FWIW, here is my experiment:

1. I have deployed Korifi on kind and pushed an app. Its workload referenced image sha `...d1e8b76e5`
2. I changed the cluster stack images:
    * `paketobuildpacks/build-jammy-base` -> `paketobuildpacks/build-jammy-full`
    * `paketobuildpacks/run-jammy-base` -> `paketobuildpacks/run-jammy-full`
3. Changing the stack immediately triggerred a new kpack build and that build produced an app image with sha `...4108c38a5`. However, the app workload image **DID NOT** change
4. I restarted the app, the workload image **DID NOT** change again
5. I restaged the app, and that picked up the new image (sha ...4108c38a5`)

**Review by danail-branekov — APPROVED** [review]

**Comment by FloThinksPi** [comment]
Thanks for testing it out @danail-branekov !

Yea i also thought about possible solutions but there is no easy one that comes to my mind straight away.
The stack seperation in CF is a unique feature that is hard to replicate in K8s and yield specific benefits(cost/resource usage) and behaviour one would lose(gurantees towards app developers). Im not aware if one can combine images on K8s at startup.

Another option is a restart waits until a implicit [cnb rebase](https://buildpacks.io/docs/for-app-developers/concepts/rebase/) is finished if a new stack shall be used and "hangs" for some time - however since the app restart is a synchronous api call that just is allowed to return when the restart was propagated correctly to the container runtime - its hard to add this in a compatible behaving way as also more errors could occur in the rebase process that need different api responses likely. However i think a transparent rebase on a restart may be the most viable option here.
